lib/libzfs/libzfs_mount.c

```diff
@@ -259,9 +259,9 @@ zfs_is_mountable(zfs_handle_t *zhp, char *buf, size_t buflen,
 
 /*
  * The filesystem is mounted by invoking the system mount utility rather
- * than by the system call mount(2).  This ensures that the /etc/mtab
+ * than by the system call mount(2).  This ensures that the /proc/self/mounts
  * file is correctly locked for the update.  Performing our own locking
- * and /etc/mtab update requires making an unsafe assumption about how
+ * and /proc/self/mounts update requires making an unsafe assumption about how
```

Comment:
Likewise, /proc/self/mounts is not updated. Perhaps these should remain /etc/mtab, because there is /etc/mtab updating that still happens?